.circleci/scripts/git-diff-develop.js

```diff
@@ -0,0 +1,68 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+const fs = require('fs');
+const path = require('path');
+
+
+async function fetchWithDepthIncrement(depth) {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error) {
+    console.error(`Failed to fetch with depth ${depth}:`, error.message);
+    return false;
+  }
+}
+
+async function gitDiffWithRetry() {
+  const depths = [5, 10, 15, 20, 30, 40, 50];
```

Comment:
to consume fewer resources, we start with small depth jumps +5 and later on we increment with +10 jumps